fraim/inputs/chunkers/base.py

```diff
@@ -0,0 +1,14 @@
+import logging
+from abc import abstractmethod
+
+
+class Chunker:
+    """Base class for chunkers."""
+
+    def __init__(self, **kwargs):
+        self.logger = kwargs.pop("logger", None)
+        if self.logger is None:
+            self.logger = logging.getLogger(__name__)
```

Comment:
I think we should require a logger in the constructor, ideally from Config

Reply:
Sounds good.

Unrelated to this PR, but more generally on this approach of passing around a logger. I might be wrong, but is it possible this design choice was due to the (more common?) approach of doing the following at the top of the file not working for debug logging?

> log = logging.getLogger(__name__)

When I was messing around on a fork I tried pulling the config object and noticed it broke debug logging. Ended up tracking it down to this issue:

https://github.com/wrale/mcp-server-tree-sitter/issues/24

Either way, separate issue, just wondering if it makes sense to pull this logger object out when it's fixed upstream.